Finished homework_3.

KMironova opened the pull request:
In src/test/java/com/epam/tc/hw3 there are pakages: ex1 for exercise 1; ex2 for exercise 2 and injectionDriver for web driver creation. Besides there are clases: ConfProperties and TestBase.
In src/main/java/com/epam/tc/hw3/pages there are clases: EpamInformationFraneworkPage and DifferentPage.
Also to src/test/resources/testing added HW3TestRun.xml and to pom.xml added one more profile setting.

KMironova committed:
Finished homework_3.
In src/test/java/com/epam/tc/hw3 there are pakages: ex1 for exercise 1; ex2 for exercise 2 and injectionDriver for web driver creation. Besides there are clases: ConfProperties and TestBase.
In src/main/java/com/epam/tc/hw3/pages there are clases: EpamInformationFraneworkPage and DifferentPage.
Also to src/test/resources/testing added HW3TestRun.xml and to pom.xml added one more profile setting.

ZeroShadows commented on src/main/java/com/epam/tc/hw3/pages/DifferentPage.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    public void verifySelectedCheckBoxWater() {
+        softAssertions.assertThat(logForCheckBoxWater.isDisplayed()).isTrue();
```
isTrue is not needed here. True for all assertThat methods, which contains isTrue

ZeroShadows commented on src/main/java/com/epam/tc/hw3/pages/EpamInformationFrameworkPage.java:
```diff
@@ -0,0 +1,202 @@
+    @FindBy(xpath = "//span[text() = 'Different elements']")
+    private WebElement differentPage;
+
+    public void openPage(String url) {
```
lets create BasePage and move this method there

ZeroShadows commented on src/main/java/com/epam/tc/hw3/pages/EpamInformationFrameworkPage.java:
```diff
@@ -0,0 +1,202 @@
+        webDriver.navigate().to(url);
+    }
+
+    public void verifyHomePageTitle(String expectedTitle) {
```
lets move common methods to the BasePage

ZeroShadows commented on src/main/java/com/epam/tc/hw3/pages/EpamInformationFrameworkPage.java:
```diff
@@ -0,0 +1,202 @@
+
+public class EpamInformationFrameworkPage {
+
+    WebDriver webDriver;
```
move WebDriver to the BaseTest and make it protected

ZeroShadows commented on src/main/java/com/epam/tc/hw3/pages/EpamInformationFrameworkPage.java:
```diff
@@ -0,0 +1,202 @@
+public class EpamInformationFrameworkPage {
+
+    WebDriver webDriver;
+    public SoftAssertions softAssertions;
```
you can initialise SoftAssertions here instead of constructor

ZeroShadows commented on src/main/java/com/epam/tc/hw3/pages/EpamInformationFrameworkPage.java:
```diff
@@ -0,0 +1,202 @@
+    }
+
+    public void verifyThatBaseIconIsDisplayed() {
+        softAssertions.assertThat(iconBase.isDisplayed()).isTrue();
```
isTrue is excessive here.
True for all assertThat where isTrue was called

ZeroShadows commented on src/test/java/com/epam/tc/hw3/injectionDriver/DriverInjector.java:
```diff
@@ -0,0 +1,6 @@
+package com.epam.tc.hw3.injectionDriver;
+
+public interface DriverInjector {
```
What is the purpose of this interfaces and classes in injectionDriver folder? I don't see any reasons to implement that complicated code

KMironova committed:
1) Deleted field injection Driver
2) Adding BasePage with common base page for pages that extends this class
3) deleted isTrue() from all methods where it was used
